src/cli/commands/force/lightning/local/setup.ts

```diff
@@ -50,40 +53,24 @@ export class Setup extends SfdxCommand {
         `sfdx force:lightning:local:setup -p Android`
     ];
 
-    public skipBaseRequirements = false;
-    public skipAdditionalRequirements = false;
-
-    private setupSteps: BaseSetup | undefined;
-
-    public async run(direct: boolean = false): Promise<any> {
-        if (direct) {
-            await this.init(); // ensure init first
-        }
+    public async run(): Promise<any> {
+        await this.init(); // ensure init first
 
         this.logger.info(`Setup command called for ${this.flags.platform}`);
 
         return this.validateInputParameters() // validate input
-            .then(() => this.executeSetup(this.setup())) // verify requirements
-            .then((result) => {
-                if (!result.hasMetAllRequirements) {
-                    return Promise.reject(
-                        new SfdxError(
-                            util.format(
-                                messages.getMessage('error:setupFailed'),
-                                this.flags.platform
-                            ),
-                            'lwc-dev-mobile-core',
-                            [
-                                messages.getMessage(
-                                    'error:setupFailed:recommendation'
-                                )
-                            ]
-                        )
-                    );
-                } else {
-                    return Promise.resolve(result);
-                }
-            });
+            .then(() =>
+                RequirementProcessor.execute(
+                    this.logger,
+                    this.commandRequirements,
+                    requirementMessages.getMessage(
+                        'error:requirementCheckFailed'
+                    ),
+                    requirementMessages.getMessage(
+                        'error:requirementCheckFailed:recommendation'
+                    )
+                )
```

Comment:
This doesn't make much sense to me. We're importing `requirementMessages` from `Requirements.ts` and then passing it back to `RequirementProcessor.execute()` in the same file. I understand that we want the `execute()` method to take messages for failure+remediation so I think we should:

1. define these 2 messages in `setup.json` message file instead and don't import `requirementMessages`

Or

2. If we suspect that the value of the 3rd and 4th parameters of `RequirementProcessor.execute()` would always be `error:requirementCheckFailed` and `error:requirementCheckFailed:recommendation` regardless of which command is calling `RequirementProcessor.execute()` then why not just remove the 3rd and 4th parameters  altogether and just have that method directly use those messages.

Reply:
`requirementMessages` is not imported but declared and initialized in `Requirements.ts`. It is also exported for use by other files(`AndroidEnvironmentRequirements` and `IOSEnvironmentRequirements`). It is referenced in `execute()` method but not passed in.

`requirement.json` is already sprinkled with resource strings used by both `AndroidEnvironmentRequirements` and `IOSEnvironmentRequirements`. Maybe this is the source of the evil 😈 

For option 1 are you suggesting that the other strings besides `error:requirementCheckFailed` and `error:requirementCheckFailed:recommendation` should be moved as well? On this note, yes, the resource file(`setup.json`) had also been a victim of it getting conflated from coupling of `Setup` and `Requirement` and needs even more careful distribution of resource strings between itself and `requirement.json` than my attempt here.

For option 2, if the caller of `RequirementProcessor.execute()` were to determine whether to show failure&recommendation messages or not we would should decide if the return value of `Promise.reject()` suffices or revive returning `SetupTestResult` so that the caller has a chance to construct failure&recommendation messages from the failed tests should he chooses to. The reason why I opted to pass failure&recommendation messages as parameters to `execute()` was because earlier we decided that we won't return `SetupTestResult` no more from `exeucte()`. And, the caller in that case should be clearly aware if any of the requirement checks fail they know what messages they want to convey to the user.

I'm open to either way. What I do want to do though and get yours and Kevin's opinion is removing the `logger` from `execute()`. The only time it is used is in `catch` handler.

Reply:
I may be wrong but I **think** when we consume -core in other repos, other commands would also make this exact call:

```
RequirementProcessor.execute(
    this.logger,
    this.commandRequirements,
    requirementMessages.getMessage(
        'error:requirementCheckFailed'
    ),
    requirementMessages.getMessage(
        'error:requirementCheckFailed:recommendation'
    )
)
```

Since every command is going to pass the same values for the 3rd & 4th param, I think we can just have `RequirementProcessor.execute(this.logger, this.commandRequirements)` and have the `execute` method internally reference those 2 messages.

I also don't want to boil the whole ocean here so take the above as just a suggestion that you can skip for this PR.

RE the `logger` I'm ok with removing it too.

Reply:
Yes, you are right, we'd have to repeat, because we always use the same failure and recommendation messages regardless of which command the requirements were checked from.

If this will keep remaining to be the case then offering the caller to specify messages may be a moot thing to do. Instead, we can just keep those messages to be shown as specified within the `execute()` function.

I made the third and fourth parameters have default values in case the caller don't want to be concerned with specifying them but offering such wiggle room may have brought this conversation. Let me remove those parameters and also clean up the resource file.

And also remove `logger` ;)